odd_collector/adapters/ckan/client.py

```diff
@@ -0,0 +1,87 @@
+import aiohttp
+from odd_collector_sdk.errors import DataSourceError
+from odd_collector.domain.plugin import CKANPlugin
+from .logger import logger
+
+from .mappers.models import Organization, Dataset, Group, ResourceField
+
+
+class CKANRestClient:
+    def __init__(self, config: CKANPlugin):
+        self.__host = f"https://{config.host}:{config.port}"
+        self.__headers = (
+            {"Authorization": config.token.get_secret_value()} if config.token else None
+        )
+
+    async def _get_request(self, url: str, params: dict = None) -> dict:
+        async with aiohttp.ClientSession(
+            self.__host, headers=self.__headers
+        ) as session:
+            try:
+                async with session.get(url, params=params) as resp:
+                    result = await resp.json()
+                    logger.debug(f"Result of request {url} is {result}")
+                return result
+            except Exception as e:
+                raise DataSourceError(
+                    f"Error during getting data from workspace {self.__host}: {e}"
+                ) from e
+
+    async def _post_request(self, url: str, payload: dict = None) -> dict:
+        async with aiohttp.ClientSession(
+            self.__host, headers=self.__headers
+        ) as session:
+            try:
+                async with session.post(url, json=payload) as resp:
+                    result = await resp.json()
+                    logger.debug(f"Result of request {url} is {result}")
+                return result
+            except Exception as e:
+                raise DataSourceError(
+                    f"Error during getting data from workspace {self.__host}: {e}"
+                ) from e
+
+    async def get_organizations(self) -> list[str]:
+        url = "/api/action/organization_list"
+        resp = await self._get_request(url)
+        if resp and resp["success"]:
```

Comment:
And may be add some logic if response was failed

Reply:
I have created the method is_response_successful and moved this check to the `_get_request`  method. In case of failure, it will raise `DataSourceError`. Only for `_post_request`, this isn't added because not all files have structure included in CKAN, and in case of failure, we just want to return an empty list.